v2/AdminCore.Services/EmployeeService.cs

```diff
@@ -122,5 +133,123 @@ private static string EncodePasswordToBase64(string password)
       return Convert.ToBase64String(inArray);
     }
+
+    private static short GetMexicanHolidays(DateTime startDate)
+    {
+      short mexicanYearOfIndependence = 1810;
+
+      return mexicanYearOfIndependence;
+    }
+
+    private static short GetNorthernIrishHolidays(DateTime startDate)
+    {
+      short holidays = 0;
+      if (startDate.Year == DateTime.Now.Year)
+      {
+        switch (startDate.Month)
+        {
+          case (int)Months.January:
+            holidays = 30;
+            break;
+
+          case (int)Months.February:
+            holidays = 28;
+            break;
+
+          case (int)Months.March:
+            holidays = 25;
+            break;
+
+          case (int)Months.April:
+            holidays = 23;
+            break;
+
+          case (int)Months.May:
+            holidays = 20;
+            break;
+
+          case (int)Months.June:
+            holidays = 18;
+            break;
+
+          case (int)Months.July:
+            holidays = 15;
+            break;
+
+          case (int)Months.August:
+            holidays = 13;
+            break;
+
+          case (int)Months.September:
+            holidays = 10;
+            break;
+
+          case (int)Months.October:
+            holidays = 8;
+            break;
+
+          case (int)Months.November:
+            holidays = 5;
+            break;
+
+          case (int)Months.December:
+            holidays = 3;
+            break;
+        }
+      }
+
+      holidays += 3;
+      return holidays;
+    }
+
+    private static bool IsNorthernIrishEmployee(Employee employee)
+    {
+      return employee.CountryId == 1;
+    }
+
+    private void AddNorthernIrishPublicHolidays(Employee employee)
+    {
+      var eventService = new EventService(DatabaseContext, _mapper, new DateService());
+      var christmasDay = new EventDateDto
+      {
+        StartDate = new DateTime(DateTime.Now.Year, (int)Months.December, 25),
+        EndDate = new DateTime(DateTime.Now.Year, (int)Months.December, 25),
+        IsHalfDay = false
+      };
+      var newYearsDay = new EventDateDto
+      {
+        StartDate = new DateTime((DateTime.Now.Year) + 1, (int)Months.January, 1),
+        EndDate = new DateTime((DateTime.Now.Year) + 1, (int)Months.January, 1),
+        IsHalfDay = false
+      };
+      var lastMondayInMay = GetLastMondayInMay();
+      var lastMondayOfMay = new EventDateDto
+      {
+        StartDate = new DateTime(DateTime.Now.Year, (int)Months.May, lastMondayInMay),
+        EndDate = new DateTime(DateTime.Now.Year, (int)Months.May, lastMondayInMay),
+        IsHalfDay = false
+      };
+
+      eventService.CreateEvent(christmasDay, EventTypes.PublicHoliday,
+        employee);
+      eventService.CreateEvent(newYearsDay, EventTypes.PublicHoliday,
+        employee);
+      eventService.CreateEvent(lastMondayOfMay, EventTypes.PublicHoliday,
+        employee);
+    }
```

Comment:
Logic looks fine, bit of a chonker method though, extract some of these object initialisations into a builder method similar to how you've done for last monday in may. Maybe consider using a loop for the create event bit at the end if it makes sense